lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -2265,8 +2275,14 @@ def has_different_resource_limits(self):
             memory=host_config.get('Memory'),
             memory_reservation=host_config.get('MemoryReservation'),
             memory_swap=host_config.get('MemorySwap'),
+            restart_policy=restart_policy_config_value
         )
 
+        # Options which don't make sense without their accompanying option
+        if self.parameters.restart_policy:
+            restart_policy = restart_policy_config_value
+            config_mapping['restart_retries'] = restart_policy.get('MaximumRetryCount')
```

Comment:
```suggestion
            config_mapping['restart_retries'] = restart_policy_config_value.get('MaximumRetryCount')
```
and remove the line above. (Or even rename `restart_policy_config_value` to `restart_policy`.)

Reply:
Done. I renamed the variable to restart_policy. Yesterday I thought it's required to distinguish between restart_policy and the new one but the existing restart_policy exists just in the scope of the config mapping dict. Tests were successful.